concepts/05 UI Components/DataGrid/20 Editing/43 Customize Edit Form/10 Appearance and Editing/20 Editor Customization.md

```diff
@@ -0,0 +1,169 @@
+    import { AppComponent } from './app.component';
+
+    import { DxDataGridModule } from 'devextreme-angular';
+
+    @NgModule({
+        declarations: [
+            AppComponent
+        ],
+        imports: [
+            BrowserModule,
+            DxDataGridModule
+        ],
+        providers: [],
+        bootstrap: [AppComponent]
+    })
+
+    export class AppModule { }
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxDataGrid ...
+            @editor-preparing="overrideOnValueChanged">
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DxDataGrid from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid
+        },
+        // ...
+        methods: {
+            overrideOnValueChanged(e) {
+                if (e.dataField === "Notes" && e.parentType === "dataRow") {
+                    e.editorName = "dxTextArea";
+                }
+                if (e.dataField === "requiredDataField" && e.parentType === "dataRow") {
+                    const defaultValueChangeHandler = e.editorOptions.onValueChanged;
+                    e.editorOptions.onValueChanged = function(args) { // override the default handler
+                        if (someCondition(e, args)) doSomething(e, args);
+                        defaultValueChangeHandler(args);
+                    }
+                }
+            }
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DataGrid from 'devextreme-react/data-grid';
+
+    const overrideOnValueChanged = (e) => {
+        if (e.dataField === "Notes" && e.parentType === "dataRow") {
+            e.editorName = "dxTextArea";
+        }
+        if (e.dataField === "requiredDataField" && e.parentType === "dataRow") {
+            const defaultValueChangeHandler = e.editorOptions.onValueChanged;
+            e.editorOptions.onValueChanged = function(args) { // override the default handler
+                if (someCondition(e, args)) doSomething(e, args);
+                defaultValueChangeHandler(args);
+            }
+        }
+    }
+
+    export default function App() {
+        return (
+            <DataGrid ... 
+                onEditorPreparing={overrideOnValueChanged}
+            >   
+            </DataGrid>
+        );
+    }
+
+---
+
+If you need to implement a custom editor for a specific data type (array, object, etc.), use the **columns[]**.[editCellTemplate](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#editCellTemplate) property. It is important to allow users to input or change the column value. Handle this value change and use the **setValue** template parameter to change the corresponding cell value. The **editCellTemplate** property specifies one template for all editing modes.
```

Comment:
> Handle this value change and use the **setValue** template parameter to change the corresponding cell value.

Do I have to handle a specific event for that purpose? If so, why don't you name the target event explicitly?

Reply:
Ok! My suggestion: Handle the **onValueChanged** event and use the **setValue** template parameter to change the corresponding cell value.

Reply:
OK. Fine by me. Thank you.